gpMgmt/bin/gpexpand

```diff
@@ -773,7 +773,18 @@ class SegmentTemplate:
 
     def _distribute_tarfile(self):
         """Distributes template tar file to hosts"""
+        localHostname = getLocalHostname()
+        # NOTE: Currently, GP's scp transfers relative file paths into the home
+        # directory by default.
+        pathsAreSame = (os.path.realpath(self.segTarFile) == os.path.join('/home/gpadmin/', self.schema_tar_file))
+
         for host in self.hosts:
+            # If the source and the destination are the same on the local
+            # machine, scp may corrupt the file. Don't overwrite it.
+            if (host == 'localhost' or host == localHostname) and pathsAreSame:
+                logger.debug('Skipping tarfile copy to %s' % host)
+                continue
```

Comment:
Could you write a test to show that we execute fewer `scp` commands?

Reply:
Sure, but it doesn't occur to me on how this could be implemeted yet. Given we need the test to see if we have avoided scp'ing the tar file on coordinator, do you have any suggestions?

Reply:
Is it possible to use this debug log?